src/makeCalculator.js

```diff
@@ -37,7 +37,33 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+    add(num) {
+      return this.result + num;
+    },
+    subtract(num) {
+      return this.result - num;
+    },
+    multiply(num) {
+      return this.result * num;
+    },
+    divide(num) {
+      return this.result / num;
+    },
+    reset(num) {
+      this.result = 0;
+
+      return this;
+    },
+    operate(callback, num) {
+      this.result = callback.call(this, num);
+
+      return this;
+    },
```

Comment:
lets make work like this for more practice with `this`
```suggestion
    operate(callback, num) {
      callback(num);

      return this;
    },
```